merge_sort.go

```diff
@@ -1,24 +1,82 @@
 package main
 
+import "fmt"
+
 type node struct {
 	v    int
 	next *node
 }
 
-func merge(a, b *node) *node {
-	// TODO
-	return nil
+func SliceToList(s []int) (n *node) {
+	for i := len(s) - 1; i >= 0; i-- {
+		n = &node{s[i], n}
+	}
+	return n
+}
+func ListToSlice(l *node) []int {
+	s := make([]int, 0)
+	for cur := l; cur != nil; cur = cur.next {
+		s = append(s, cur.v)
+	}
+	return s
+}
+func Print(l *node) {
+	for i := l; i != nil; i = i.next {
+		fmt.Println(i.v)
+	}
+	fmt.Println()
+}
```

Comment:
Given that these functions are used for tests only, they should be defined in `merge_sort_test.go`.